x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/string/InsertFunctionProcessor.java

```diff
@@ -67,8 +67,15 @@ public static Object doProcess(Object input, Object start, Object length, Object
         if (!(length instanceof Number)) {
             throw new SqlIllegalArgumentException("A number is required; received [{}]", length);
         }
-        if (((Number) length).intValue() < 0) {
-            throw new SqlIllegalArgumentException("A positive number is required for [length]; received [{}]", length);
+        
+        if ((((Number) length).longValue() > Integer.MAX_VALUE) || (((Number) length).longValue() < 0) ) {
+            throw new SqlIllegalArgumentException("[length] must be in the interval [0..2147483647], but was ["
+                    + ((Number) length).longValue() + "]");
+        }
+        
+        if ((((Number) start).longValue() > Integer.MAX_VALUE) || (((Number) start).longValue() - 1 < Integer.MIN_VALUE)) {
```

Comment:
Please check the arguments in their order, so first the `start` then the `length`.